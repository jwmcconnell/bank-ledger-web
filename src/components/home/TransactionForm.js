import React from 'react';
import PropTypes from 'prop-types';

const TransactionForm = ({ handleUpdate, handleSubmit, depositValue, transactionType }) => {
  return (
    <form onSubmit={(e) => handleSubmit(e, transactionType)}>
      <label>{transactionType === 'deposit' ? 'Deposit' : 'Withdrawal'} Amount:
        <input name={transactionType} type="number" step="0.01" onChange={handleUpdate} value={depositValue}/>
      </label>
      <button>Submit</button>
    </form>
  );
};

TransactionForm.propTypes = {
  handleUpdate: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  depositValue: PropTypes.number.isRequired,
  transactionType: PropTypes.string.isRequired
};

export default TransactionForm;
