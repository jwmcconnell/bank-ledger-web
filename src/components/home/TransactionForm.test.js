import React from 'react';
import { shallow } from 'enzyme';
import TransactionForm from './TransactionForm';

describe('TransactionForm component', () => {
  it('renders TransactionForm', () => {
    const wrapper = shallow(<TransactionForm 
      handleSubmit={() => {}} 
      handleUpdate={() => {}}
      depositValue={123}
      transactionType='withdrawal'
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
