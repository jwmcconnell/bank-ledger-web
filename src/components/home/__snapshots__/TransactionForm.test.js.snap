// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionForm component renders TransactionForm 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Withdrawal
     Amount:
    <input
      name="withdrawal"
      onChange={[Function]}
      step="0.01"
      type="number"
      value={123}
    />
  </label>
  <button>
    Submit
  </button>
</form>
`;
